test/functional/forall/reduce-sanity/CMakeLists.txt

```diff
@@ -5,36 +5,63 @@
 # SPDX-License-Identifier: (BSD-3-Clause)
 ###############################################################################
 
-raja_add_test(
-  NAME test-forall-reduce-sanity-seq
-  SOURCES test-forall-reduce-sanity-seq.cpp)
+#
+# Function that generates test file and build target for each
+# reduction type and each backend
+#
+function( buildforallreducesanitytest REDUCETYPES )
+  foreach( REDUCETYPE ${REDUCETYPES} )
+    configure_file( test-forall-reduce-sanity.cpp.in 
+                    test-forall-${REDUCETYPE}-sanity-${BACKEND}.cpp )
+    raja_add_test( NAME test-forall-${REDUCETYPE}-sanity-${BACKEND}
+                   SOURCES test-forall-${REDUCETYPE}-sanity-${BACKEND}.cpp )
+
+    target_include_directories(test-forall-${REDUCETYPE}-sanity-${BACKEND}.exe 
+                               PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/tests)
+  endforeach()
+endfunction()
+
+
+#
+# Variable holding reduction types
+#
+set(REDUCETYPES ReduceSum ReduceMin ReduceMax ReduceMinLoc ReduceMaxLoc)
+
+
+#
+# Generate tests for each enabled RAJA back-end
+#
+
+set( BACKEND Sequential )
```

Comment:
@MrBurmark is this what you had in mind?

Reply:
Ya that's what I was thinking.

Reply:
@MrBurmark do you see any other opportunities to cut things down? I think it's just about as lean as we can make it. Adding a test for a new back-end will now require a couple of lines added to the CMakeLists.txt file.